providers/heyreach/handlers.go

```
@@ -0,0 +1,68 @@
+package heyreach
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
+
+func (c *Connector) buildReadRequest(ctx context.Context, params common.ReadParams) (*http.Request, error) {
+	urlPath, err := matchReadObjectNameToEndpointPath(params.ObjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := urlbuilder.New(c.ProviderInfo().BaseURL, "public", urlPath)
+	if err != nil {
+		return nil, err
+	}
+
+	body := constructRequestBody(params)
+
+	jsonData, err := json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+
+	return http.NewRequestWithContext(ctx, http.MethodPost, url.String(), bytes.NewReader(jsonData))
+}
+
+func (c *Connector) parseReadResponse(
+	ctx context.Context,
+	params common.ReadParams,
+	request *http.Request,
+	response *common.JSONHTTPResponse,
+) (*common.ReadResult, error) {
+	offset, _ := strconv.Atoi(params.NextPage.String())
```

I think it is beneficial to propagate an error to notify the user that NextPage their specified is invalid, otherwise we silently return the first page which could be not the intention.